tools/run/solo5-virtio-run.sh

```diff
@@ -0,0 +1,186 @@
+        shift; shift
+        ;;
+    -H)
+        HV="$2"
+        shift; shift
+        ;;
+    --)
+        shift; break
+        ;;
+    esac
+done
+[ $# -lt 1 ] && usage
+
+UNIKERNEL=$(readlink -f $1)
+shift
+[ ! -f ${UNIKERNEL} ] && die "not found: ${UNIKERNEL}"
+VMNAME=$(basename ${UNIKERNEL})
+
+if [ "${HV}" = "best" ]; then
+    SYS=$(uname -s)
+    case ${SYS} in
+    Linux)
+        if [ -c /dev/kvm -a -w /dev/kvm ]; then
+            HV=kvm
+        else
+            HV=qemu
+        fi
+        ;;
+    FreeBSD)
+        # XXX How to detect if bhyve is available on this machine?
+        HV=bhyve
+        ;;
+    *)
+        die "unsupported os: ${SYS}"
+        ;;
+    esac
+fi
+
+case ${HV} in
+kvm|qemu)
+    hv_addargs qemu-system-x86_64
+    case ${HV} in
+    kvm)
+        hv_addargs -cpu host -enable-kvm
+        ;;
+    qemu)
+        hv_addargs -cpu Westmere
+        ;;
+    esac
+    hv_addargs -m ${MEM}
+
+    # Kill all default devices provided by QEMU, we don't need them.
+    hv_addargs -nodefaults -no-acpi
+
+    # Console. We could use just "-nograhic -vga none", however that MUXes the
+    # QEMU monitor on stdio which requires ^Ax to exit. This makes things look
+    # more like a normal process (quit with ^C), consistent with bhyve and ukvm.
+    hv_addargs -display none -serial stdio
+
+    # Network
+    if [ -n "${NETIF}" ]; then
+        hv_addargs -device virtio-net,netdev=n0
+        hv_addargs -netdev tap,id=n0,ifname=${NETIF},script=no,downscript=no
+    fi
+    # Disk
+    if [ -n "${BLKIMG}" ]; then
+        hv_addargs -drive file=${BLKIMG},if=virtio,format=raw
+    fi
+
+    hv_addargs -kernel ${UNIKERNEL}
+
+    set -x
+    exec ${HVCMD}
+    ;;
+bhyve)
+    # Just blow away any pre-existing VM of the same name.
+    bhyvectl --destroy --vm=${VMNAME} >/dev/null 2>&1
+    # Load the VM using grub-bhyve. Kill stdout as this is normal GRUB output.
+    (set -x; \
+        printf "multiboot ${UNIKERNEL} $@\nboot\n" \
+        | grub-bhyve -M ${MEM} "${VMNAME}" >/dev/null) \
+        || die "Could not initialise VM"
+
+    hv_addargs bhyve
+    hv_addargs -m ${MEM}
+    hv_addargs -H -s 0:0,hostbridge -s 1:0,lpc
+
+    # Console. Bhuve insists on talking to a TTY so fake one in a way that we
+    # can redirect output to a file (see below).
+    TTYA=/dev/nmdm$$A
+    TTYB=/dev/nmdm$$B
+    hv_addargs -l com1,${TTYB}
+    
+    # Network
+    if [ -n "${NETIF}" ]; then
+        hv_addargs -s 2:0,virtio-net,${NETIF}
+    fi
+    # Disk
+    if [ -n "${BLKIMG}" ]; then
+        hv_addargs -s 3:0,virtio-blk,${BLKIMG}
+    fi
+
+    hv_addargs ${VMNAME}
+
+    # Fake a console using nmdm(4). Open 'A' end first and keep it open, then
+    # set some sane TTY parameters and launch bhyve on the 'B' end.
+    # Bhyve and cat will respond to interactive SIGINT correctly.
+    # XXX Can occasionally leak /dev/nmdm devices, oh well ...
+    cat ${TTYA} &
```

Comment:
presumably you want the console output to go somewhere in any case? if not you can always `>/dev/null`